libs/ledger/src/storage_unit/transaction_store_sync_service.cpp

```diff
@@ -106,32 +98,30 @@ TransactionStoreSyncService::TransactionStoreSyncService(Config const &cfg, Mudd
   , current_tss_peers_{telemetry::Registry::Instance().CreateGauge<uint64_t>(
         "current_tss_peers", "The number of peers the sync can use")}
 {
-  state_machine_->RegisterHandler(State::INITIAL, this, &TransactionStoreSyncService::OnInitial);
-  state_machine_->RegisterHandler(State::QUERY_OBJECT_COUNTS, this,
-                                  &TransactionStoreSyncService::OnQueryObjectCounts);
-  state_machine_->RegisterHandler(State::RESOLVING_OBJECT_COUNTS, this,
-                                  &TransactionStoreSyncService::OnResolvingObjectCounts);
-  state_machine_->RegisterHandler(State::QUERY_SUBTREE, this,
-                                  &TransactionStoreSyncService::OnQuerySubtree);
-  state_machine_->RegisterHandler(State::RESOLVING_SUBTREE, this,
-                                  &TransactionStoreSyncService::OnResolvingSubtree);
-  state_machine_->RegisterHandler(State::QUERY_OBJECTS, this,
-                                  &TransactionStoreSyncService::OnQueryObjects);
-  state_machine_->RegisterHandler(State::RESOLVING_OBJECTS, this,
-                                  &TransactionStoreSyncService::OnResolvingObjects);
-  state_machine_->RegisterHandler(State::TRIM_CACHE, this,
-                                  &TransactionStoreSyncService::OnTrimCache);
+  // clang-format off
```

Comment:
Lets use the previous version (maybe introduce the clang format off and tidy a little. The main reason is that this way are are limited to using signature: `State(TransactionStoreSyncService::*)()` however, it is very useful when debugging to be able to change this to one of the other handlers like: `State(TransactionStoreSyncService::*)(State, State)`